src/main/java/randoop/generation/GrtImpurity.java

```diff
@@ -0,0 +1,513 @@
+package randoop.generation;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import randoop.main.GenInputsAbstract;
+import randoop.main.RandoopBug;
+import randoop.operation.CallableOperation;
+import randoop.operation.ConstructorCall;
+import randoop.operation.MethodCall;
+import randoop.operation.TypedClassOperation;
+import randoop.operation.TypedOperation;
+import randoop.sequence.Sequence;
+import randoop.types.NonParameterizedType;
+import randoop.types.Type;
+import randoop.types.TypeTuple;
+import randoop.util.Randomness;
+
+/**
+ * Implements the "GRT Impurity" component, as described in "GRT: Program-Analysis-Guided Random
+ * Testing" by Ma et. al (ASE 2015): https://people.kth.se/~artho/papers/lei-ase2015.pdf.
+ *
+ * <p>The GRT Impurity component is a fuzzing mechanism that alters the states of (or creates new)
```

Comment:
This is an accurate comment.  Since it is a fuzzing mechanism, I suggest renaming this to "GRT Fuzzing" instead of "GRT Immutability".  This introduces an unfortunate terminology difference with the paper, but the paper's terminology is terrible.  This class should clearly state that it is an implementation of whan is called "Immutability" in the GRT paper.